### PR TITLE
Release v0.21.0: refresh website and publish DeltaSelect guidance

### PR DESCRIPTION
**Pull request by nicholasjconn**
## Summary

Refresh the website around Agent Dispatch, Git-backed skill imports, and DeltaSelect benchmark selection, and prepare the approved v0.21.0 minor release.

## Changes

- Publish the DeltaSelect landing page, interactive tool, and verified arXiv 2609.19607 citation; preserve the previous planner URL with a redirect.
- Add the optional benchmark guidance skill, `llms.txt`, and source Markdown documentation mirrors.
- Correct provider-isolation guidance and use a synthetic website CI version that survives retention.
- Add v0.21.0 release notes, upgrade documentation, and migration and template ownership manifests.

## Verification

- Full local CI passed before release preparation, including 850 end-to-end assertions and six upgrade scenarios.
- Production website build for v0.21.0 passed from a fresh website checkout.
- Browser navigation confirmed that /deepswe-planner redirects to /deltaselect-tool and loads the planner.
- `make release-preflight RELEASE_TAG=v0.21.0` — passed, including 917 end-to-end assertions, eight upgrade scenarios, and release manifest/documentation validation.

### CODE REVIEW

File: site/pages/deltaselect-tool.jsx [hunk 1]
```diff
@@ -40,20 +40,20 @@ function PlannerFrame() {
...
- * documentation theme styles while preserving same-origin clipboard access.
+ * Render the self-contained DeltaSelect benchmark task-selection tool inside
+ * the website shell while keeping its validated evidence layout isolated from
+ * documentation theme styles.
  *
- * @returns {React.JSX.Element} website task-evidence page
+ * @returns {React.JSX.Element} DeltaSelect tool page
  */
-export default function DeepSWEPlannerPage() {
+export default function DeltaSelectToolPage() {
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply site/pages/deltaselect-tool.jsx:49]: Fixed in f1d180868b46da26c767b61c105a55dbf65c3c9d. Restored /deepswe-planner as a redirect to /deltaselect-tool. The production build passes, and browser navigation from the legacy URL reaches the new page and loads the planner iframe.

File: README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Key properties:
 - Local-first, no telemetry
 - Deterministic outputs from canonical inputs
 - Explicit approvals and command allowlists
-- Optional per-repo runtime isolation (Codex auth/sessions/logs; Claude settings and caches with auth pending upstream fix)
+- Optional per-repo runtime isolation (Codex auth, sessions, and logs; Claude settings, caches, and `/login` credentials)
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply README.md:17]: Fixed in f1d180868b46da26c767b61c105a55dbf65c3c9d. Reconciled the FAQ with the other public guidance. Inspection of installed Claude Code 2.1.274 confirms the macOS service name includes a hash of the configuration directory; the FAQ also calls out the explicit secure-storage override and external authentication sources.

File: site/docs/concepts.mdx [hunk 3]
```diff
@@ -99,7 +99,7 @@ If you find yourself carrying many permanent overrides, that is usually a sign t
 
 ## Per-repo isolation {#per-repo-credential-isolation}
 
-Agent Layer gives each repository its own agent configuration so work in one repo never leaks into another. Codex home isolation is opt-in. Claude settings and caches isolation is supported; whether it also isolates login credentials depends on your platform and authentication method (see below).
+Agent Layer gives each repository its own agent configuration so work in one repo never leaks into another. Codex home isolation is opt-in. Claude settings, caches, and `/login` credential isolation are also opt-in; other authentication modes may still use external credential sources.
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply site/docs/concepts.mdx]: Fixed in f1d180868b46da26c767b61c105a55dbf65c3c9d. Limited the claim to Agent Layer-managed configuration and explicitly stated that provider configuration and runtime state can remain shared.

File: site/pages/faq.mdx [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
 
 ## What problem does Agent Layer solve?
+
 It keeps instructions, skills, MCP servers, and approvals consistent across multiple agents by generating every client config from `.agent-layer/`.
 
 If you run more than one client, you edit that folder once and review the same files in git.
 
+## Can one coding agent delegate work to another provider?
+
+Yes. [Agent Dispatch](/docs/agent-dispatch) lets any supported caller start asynchronous Claude, Codex, Grok, or Antigravity conversations through built-in MCP tools. People and automation can use matching `al dispatch` CLI commands. The caller receives a handle immediately, waits when it needs the result, and can continue the same provider conversation later.
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply site/pages/faq.mdx]: Fixed in f1d180868b46da26c767b61c105a55dbf65c3c9d. Qualified MCP callers by working MCP support, preserved Antigravity as a dispatch target, and documented the current probe limitation plus al probe agy.

File: cmd/publish-site/main.go [hunk 5]
```diff
@@ -158,6 +163,53 @@ func publishDeepSWEPlanner(repoA, repoB string) error {
...
+		return err
+	}
+	return filepathWalkFunc(destination, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() || filepath.Ext(path) != ".mdx" {
+			return nil
+		}
+		return os.Rename(path, strings.TrimSuffix(path, ".mdx")+".md")
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply cmd/publish-site/main.go:209]: Disagreed. These are intentionally untransformed source-document mirrors, not a separate Markdown navigation system. The relative links resolve to valid server-rendered documentation, and llms.txt enumerates the direct .md endpoints for text clients. Rewriting links would add a transformation contract beyond the source-mirror behavior documented and tested in this PR.

**Comment by nicholasjconn** [comment]
Disagreed. Regarding the docstring coverage warning in https://github.com/conn-castle/agent-layer/pull/229#issuecomment-5724828994: the new publication helpers already explain their ownership and source-mirroring contracts. Adding docstrings to self-describing test cases solely to satisfy this percentage would duplicate their names without improving behavioral documentation. Repository lint, coverage, and CI gates passed; no repository policy requires this separate bot percentage.

File: site/pages/deltaselect.mdx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+DeltaSelect is for developers who need to measure whether an instruction, Agent Skill, model, harness, or workflow change actually helps. Run selected tasks through coding-agent subscriptions you already pay for, instead of spending thousands of dollars on full benchmark runs through custom harnesses. Use it to plan a smaller evaluation with a clear budget, a fixed task set, and visible limits on what the result can tell you.
+
+<div className="hero-buttons">
+  <Link to="/deltaselect-tool" className="hero-button hero-button--primary">Open DeltaSelect</Link>
+  <Link to="/docs/reference#benchmark-runner" className="hero-button hero-button--secondary">Benchmark runner</Link>
+</div>
+
+:::note[Research paper]
+The [DeltaSelect arXiv paper](https://arxiv.org/abs/XXXX.XXXXX) describes the method, validation, and limitations. DeltaSelect is available in Agent Layer v0.18.0.
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply site/pages/deltaselect.mdx]: Fixed in 05b5658bbd54b4b901340929b9adea6d36c59aff. Both public paper links now use https://arxiv.org/abs/2609.19607, and both BibTeX entries use the verified published title, author, year, and identifier. Removed the provisional-citation notices.

[the remark this answers is not in the document]
Comment (nicholasjconn) [reply site/pages/deltaselect.mdx]: Fixed in 05b5658bbd54b4b901340929b9adea6d36c59aff. Both public paper links now use https://arxiv.org/abs/2609.19607, and both BibTeX entries use the verified published title, author, year, and identifier. Removed the provisional-citation notices.

File: site/static/deepswe-planner/app/index.html [hunk 7]
```diff
@@ -805,6 +844,11 @@ <h1>A cost-effective benchmark for instruction and skills</h1>
         selects 9 tasks.
       </p>
     </div>
+    <!-- Replace the paper URL with the published arXiv URL before release. -->
+    <aside class="publication-card" aria-labelledby="publication-title">
+      <h2 id="publication-title">Research paper</h2>
+      <p>Read the <a href="https://arxiv.org/abs/XXXX.XXXXX" target="_blank" rel="noopener noreferrer">DeltaSelect arXiv paper</a> for a complete description of the method, evaluation, and limitations.</p>
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply site/static/deepswe-planner/app/index.html]: Fixed in 05b5658bbd54b4b901340929b9adea6d36c59aff. Both public paper links now use https://arxiv.org/abs/2609.19607, and both BibTeX entries use the verified published title, author, year, and identifier. Removed the provisional-citation notices.

File: site/static/llms.txt [hunk 8]
```diff
@@ -0,0 +1,29 @@
+# Agent Layer
+
+> Agent Layer is a repo-local control plane for coding agents. It provides asynchronous cross-provider Agent Dispatch, Git-backed portable Agent Skills, shared MCP and approval configuration, and DeltaSelect benchmark selection.
+
+The canonical source repository is https://github.com/conn-castle/agent-layer. Documentation pages are server-rendered for browsers and also published as Markdown at the `.md` URLs below for agents and text-only clients.
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply site/static/llms.txt]: Fixed in 05b5658bbd54b4b901340929b9adea6d36c59aff. llms.txt now explicitly limits the Markdown mirror to /docs reference documentation and distinguishes the remaining server-rendered page links.

**Comment by nicholasjconn** [comment]
Fixed in 05b5658bbd54b4b901340929b9adea6d36c59aff. Addressed the remaining documentation findings in https://github.com/conn-castle/agent-layer/pull/229#pullrequestreview-5244069349:

- Published the verified arXiv identifier and citation on both DeltaSelect surfaces.
- Corrected the minimum compatible CLI version to v0.18.4 and documented the selection export followed by CLI initialization. The tool also has a copy-study-template action, but the initialization route is the recommended self-contained workflow.
- Removed references to the missing GitHub and Grok image assets, retaining the GitHub link and a text Grok label.
- Narrowed the llms.txt coverage claim to the mirrored /docs documents.
- Updated the stale credential decision and benchmark memory. Installed Claude Code 2.1.274 derives the macOS Keychain service suffix from a SHA-256 hash of the configuration directory; the FAQ records the secure-storage-directory override and external authentication caveats.

The relative-link transformation recommendation was assessed separately in https://github.com/conn-castle/agent-layer/pull/229#discussion_r4043658598. These are intentionally source mirrors; their direct Markdown URLs are indexed by llms.txt. CodeRabbit withdrew the same finding after reviewing that contract.

**Comment by nicholasjconn** [comment]
Disagreed. The repeated Markdown-navigation warning in https://github.com/conn-castle/agent-layer/pull/229#issuecomment-5724828994 is already covered by the source-mirror contract and was withdrawn in https://github.com/conn-castle/agent-layer/pull/229#discussion_r4043675789. I additionally checked the fresh v0.21.0 production build: all 126 relative Markdown links resolve to generated rendered pages, with zero missing targets. llms.txt indexes the direct .md sources for text clients; the mirror deliberately preserves source content rather than promising an entirely separate Markdown navigation graph. No rewrite is needed.
